### PR TITLE
docs: update guides and capitalization [JOB-61191]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

As our casing for titles changes from UPPERCASE to
- sentence-case for headings
- titlecase for button labels

Our content guidance should be updated to reflect this, and our documentation should follow the same patterns we describe.

### Added

- new capitalization section to our content docs
- capitalization guidelines to components like Heading and Button

### Changed

- updated casing of common section headings (Content guidelines, Design & usage guidelines, etc)
- casing of headers in main repo README and contributing guides

## Testing

View the associated docs pages and confirm guidelines exist and section headings follow appropriate casing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: docs/content/voice-and-tone.mdx [hunk 1]
```diff
@@ -27,13 +27,24 @@ do).
 
 ### Punctuation
 
+#### Periods
+
```
Comment (chris-at-jobber) [inline docs/content/voice-and-tone.mdx:31]: @agglasaur we already had punctuation in this doc so I added ampersand guidance here - alternatively, could be worth moving all punctuation to formatting?

Comment (agglasaur) [reply docs/content/voice-and-tone.mdx:31]: Yeah probably makes sense in formatting, that would be my first place to look

File: packages/components/src/Heading/Heading.mdx [hunk 2]
```diff
@@ -103,6 +103,19 @@ of content.
 
 ---
 
+## Content guidelines
+
+The only content that should be used in a `Heading` is text. Use sentence-case
+for headings. The exception is `Heading 6`, which is capitalized as a stylistic
+reference to the
+[eyebrow typgraphic pattern.](https://app.uxcel.com/lessons/basics-ii-588#eyebrow-headline-8095)
```
Comment (agglasaur) [inline packages/components/src/Heading/Heading.mdx:111]: never actually heard this expression, but nothing has ever made more sense

File: packages/components/src/DescriptionList/DescriptionList.mdx [hunk 3]
```diff
@@ -34,7 +34,7 @@ import { DescriptionList } from "@jobber/components/DescriptionList";
 
 <Props of={DescriptionList} />
 
-## Design & Usage Guidelines
+## Design & usage guidelines
```
Comment (MichaelParadis) [inline packages/components/src/DescriptionList/DescriptionList.mdx:37]: Header below should be `Passing an element` not `Passing an Element`

File: packages/components/src/DatePicker/DatePicker.mdx [hunk 4]
```diff
@@ -41,7 +41,7 @@ import { DatePicker } from "@jobber/components/DatePicker";
 
 <Props of={DatePicker} />
 
-## Design & Usage Guidelines
+## Design & usage guidelines
```
Comment (MichaelParadis) [inline packages/components/src/DatePicker/DatePicker.mdx:44]: Should the `Custom Activator` heading be updated too?

File: packages/components/src/DataTable/DataTable.mdx [hunk 5]
```diff
@@ -1023,7 +1023,7 @@ content?
   />
 </Playground>
 
-## Design & Usage Guidelines
+## Design & usage guidelines
```
Comment (MichaelParadis) [inline packages/components/src/DataTable/DataTable.mdx:1026]: Some of the other headings in this file don't follow the casing guidelines.

File: packages/components/src/ConfirmationModal/ConfirmationModal.mdx [hunk 6]
```diff
@@ -46,7 +46,7 @@ import {
   }}
 </Playground>
 
-## Design & Usage Guidelines
+## Design & usage guidelines
```
Comment (MichaelParadis) [inline packages/components/src/ConfirmationModal/ConfirmationModal.mdx:49]: `Controlled Confirmation Modal` should be updated.

**Review by MichaelParadis — COMMENTED** [review]
Minor comments but overall looks good. I think I will see if I can enforce this at a linter level but I don't think this should block these changes.

The update to the generator also reduces the chance this isn't followed because it creates the headers it for you

File: packages/components/src/Page/Page.mdx [hunk 7]
```diff
@@ -63,7 +63,7 @@ otherwise does not benefit from horizontal constraints.
 
 ---
 
-## Page with Actions
+## Page with qctions
```
Comment (MichaelParadis) [inline packages/components/src/Page/Page.mdx:66]: `actions`

Comment (chris-at-jobber) [reply packages/components/src/Page/Page.mdx:66]: 🤦

Comment (chris-at-jobber) [reply packages/components/src/Page/Page.mdx:66]:
```suggestion
## Page with actions
```

**Review by agglasaur — APPROVED** [review]

**Pull request merged**